autogpt_platform/frontend/src/hooks/useAgentGraph.ts

```diff
@@ -405,7 +405,34 @@ export default function useAgentGraph(
             setNestedProperty(errors, key, "This field is required");
           }
         });
-      }
+      } 
+      
+      Object.entries(node.data.inputSchema.properties || {}).forEach(([key, schema]) => {
+        if ('depends_on' in schema) {
+          const dependencies = Array.isArray(schema.depends_on) ? schema.depends_on : [schema.depends_on];
+          
+          // Check if dependent field is set while required fields are empty
+          const hasValue = inputData[key as keyof typeof inputData] && String(inputData[key as keyof typeof inputData]).length > 0;
+          const missingDependencies = dependencies.filter(dep => !inputData[dep as keyof typeof inputData] || String(inputData[dep as keyof typeof inputData]).length === 0);
+          
+          if (hasValue && missingDependencies.length > 0) {
+            errors[key] = `Requires ${missingDependencies.join(', ')} to be set`;
+            errorMessage = `Field ${key} requires ${missingDependencies.join(', ')} to be set`;
+          }
+          
+          // Check if required fields are set but dependent field is empty
+          const hasAllDependencies = dependencies.every(dep => 
+            inputData[dep as keyof typeof inputData] && 
+            String(inputData[dep as keyof typeof inputData]).length > 0
+          );
+          if (hasAllDependencies && (!inputData[key as keyof typeof inputData] || String(inputData[key as keyof typeof inputData]).length === 0)) {
+            errors[key] = `This field is required when ${dependencies.join(', ')} are set`;
+            errorMessage = `${key} is required when ${dependencies.join(', ')} are set`;
+          }
```

Comment:
`key as keyof typeof inputData`

Is this necessary?

Reply:
It’s giving me a TS error, which is why I added it.